### PR TITLE
Add ACME/Let’s Encrypt integration tests

### PR DESCRIPTION
**Pull request by trecloux**
Tests are based on a boulder image I pushed on my personal docker hub account.

Thx @gwallet for the help.

### CODE REVIEW

File: integration/acme_test.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	"github.com/containous/traefik/integration/utils"
+	checker "github.com/vdemeester/shakers"
+)
+
+// Consul test suites (using libcompose)
+type AcmeSuite struct {
+	BaseSuite
+}
+
+func (s *AcmeSuite) setupBoulder(c *check.C) {
```
Comment (vdemeester) [inline integration/acme_test.go]: This could be in `SetUpSuite` and not called by the test I think :angel:

File: integration/integration_test.go [hunk 2]
```diff
@@ -52,6 +54,18 @@ func (s *BaseSuite) TearDownSuite(c *check.C) {
 func (s *BaseSuite) createComposeProject(c *check.C, name string) {
 	projectName := fmt.Sprintf("integration-test-%s", name)
 	composeFile := fmt.Sprintf("resources/compose/%s.yml", name)
+
+	addrs, err := net.InterfaceAddrs()
```
Comment (vdemeester) [inline integration/integration_test.go]: @trecloux care to explain this change ? :angel:

File: integration/acme_test.go (hunk 1 above)
Comment (trecloux) [inline integration/acme_test.go]: I used the same structure than the ConsulSuite, but yes sounds good :-)

File: integration/integration_test.go (hunk 2 above)
Comment (vdemeester) [inline integration/integration_test.go]: ah nevermind, didn't look at the compose file :joy:

File: integration/acme_test.go (hunk 1 above)
Comment (trecloux) [inline integration/acme_test.go]: Fixed

File: integration/resources/compose/boulder.yml [hunk 3]
```diff
@@ -0,0 +1,44 @@
+boulder:
+    image: trecloux/boulder:20161209-8a6382f4
```
Comment (dtomcej) [inline integration/resources/compose/boulder.yml]: What changes have been made to `letsencrypt/boulder`?

Comment (trecloux) [inline integration/resources/compose/boulder.yml]: No change, I just built/pushed the last commit of the release branch 8a6382f4 is the commit id. The official image was not updated for 9 months.

**Comment by emilevauge** [comment]
@trecloux If you don't mind, could you use this image instead:
https://hub.docker.com/r/containous/boulder automatically build from https://github.com/containous/boulder ?

**Comment by trecloux** [comment]
@emilevauge I updated the image. Unfortunately the test looks instable on travis

File: integration/acme_test.go [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+
+	// wait for traefik (generating acme account take some seconds)
+	err = utils.Try(30*time.Second, func() error {
+		_, err := http.Get("https://127.0.0.1:5001")
+		if err != nil {
+			return err
+		}
+		return nil
+	})
+
```
Comment (emilevauge) [inline integration/acme_test.go]: You should add `c.Assert(err, checker.IsNil)` here I think ;)

**Comment by trecloux** [comment]
@emilevauge Thanks for the tip, fixed it.
@vdemeester Ok for you ?

**Comment by trecloux** [comment]
@vdemeester I rebased the branch, should I squash the commits ?

**Comment by emilevauge** [comment]
@trecloux 
>should I squash the commits ?

In this case yeah, we don't need to keep `Copy paste is bad :-)` commits in history ;)

**Pull request merged**